src/actions/RemoveSelection.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  TypedSelection,
+  Graph,
+} from "../typings/Types";
+import { runOnTargetsForEachEditor } from "../util/targetUtils";
+
+export default class RemoveSelection implements Action {
+  getTargetPreferences: () => ActionPreferences[] = () => [
+    { insideOutsideType: "inside" },
+  ];
+
+  constructor(private graph: Graph) {
+    this.run = this.run.bind(this);
+  }
+
+  protected getSelection(target: TypedSelection) {
+    return target.selection.selection;
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    await runOnTargetsForEachEditor(targets, async (editor, targets) => {
+      editor.selections = editor.selections.filter(
+        (selection) =>
+          !targets.find(
+            (target) =>
+              target.selection.selection.intersection(selection) &&
+              !target.selection.selection.start.isEqual(selection.end) &&
+              !target.selection.selection.end.isEqual(selection.start)
```

Comment:
I believe you're checking that the selection has non empty overlap? Can't you just ask the intersection whether it is empty? If not, I'd add a comment to clarify what this is doing

Reply:
Well you could do that. . . if it wasn't in the middle of the night :p
I have updated it now :)